include/odp/api/spec/queue_types.h

```diff
@@ -291,6 +301,29 @@ typedef struct odp_queue_param_t {
 	  * default size. The default value is 0. */
 	uint32_t size;
 
+	/** Number of vector aggregators
+	  *
+	  * When >= 1, event vector generation is enabled and configured
+	  * according to vector configuration parameters. Otherwise, vector
+	  * configuration parameters are ignored. The default value is 0.
+	  *
+	  * Depending on queue type, vector aggregation capabilities can be read
+	  * either from odp_queue_capability_t::plain.vector or
+	  * odp_schedule_capability_t::vector. */
+	uint32_t num_vector;
+
+	/** Vector configuration parameters
+	  *
+	  * When event vector generation is enabled, events enqueued to this
+	  * queue are by default vectorized using vector aggregator 0. When
+	  * 'num_vector' >1, other vector aggregators can be utilized with
+	  * odp_queue_enq_vector() function.
+	  *
+	  * When vector generation is enabled, events may be delivered both as
+	  * event vector events and normal events. Implementation won't split up
+	  * event vectors successfully enqueued by the application. */
+	odp_event_vector_config_t vector[ODP_QUEUE_MAX_VECTOR_AGGR];
```

Comment:
Having multiple vector aggregators per queue adds flexibility and might be useful for decoupling vector aggregation from scheduling contexts (i.e. one could aggregate pktio and crypto packets separately but still have them go through the same scheduled queue, e.g. for atomicity).

However, configuring the aggregators this way in a fixed manner at queue creation time is rather inflexible. It would not be possible to e.g. have a new aggregator for a newly created crypto session attached to an existing queue. If we want to support multiple aggregators per queue, I think it would be nice to be able to create the aggregators dynamically or at least associate them to queues dynamically.